src/Components/Components/src/ICascadingValueSupplierFactory.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal interface ICascadingValueSupplierFactory
+{
+    bool TryGetValueSupplier(object propertyAttribute, Type valueType, string? valueName, [NotNullWhen(true)] out ICascadingValueSupplier? result);
+}
```

Comment:
This functionality was pulled into its own interface to allow multiple cascading values to be associated with a single source. Currently, `CascadingModelBinder` and `CascadingValue<T>` implement both these interfaces, but my thinking is that `CascadingModelBinder` will change to only implement `ICascadingValueSupplierFactory`.

Similarly, I'm also working on a new `CascadingQueryValueProvider` component that creates an `ICascadingValueSupplier` per query parameter.

Reply:
That makes sense.